android/codepush.gradle

```diff
@@ -42,40 +49,38 @@ gradle.projectsEvaluated {
         def jsBundleFile;
 
         // Additional node commandline arguments
-        def nodeExecutableAndArgs = config.nodeExecutableAndArgs ?: ["node"]
-        def extraPackagerArgs = config.extraPackagerArgs ?: []
+        def nodeExecutableAndArgs = config.nodeExecutableAndArgs.get() ?: ["node"]
+        def extraPackagerArgs = config.extraPackagerArgs.get() ?: []
 
         // Make this task run right after the bundle task
         def generateBundledResourcesHash;
 
-        if (variant.hasProperty("bundleJsAndAssets")) {
-            def reactBundleTask = variant.bundleJsAndAssets
-            jsBundleDir = reactBundleTask.generatedAssetsFolders[0].absolutePath
-            resourcesDir = reactBundleTask.generatedResFolders[0].absolutePath
+        def reactBundleTask = tasks.findByName("createBundle${targetName}JsAndAssets")
+        if (reactBundleTask) {
+            jsBundleDir = reactBundleTask.property('jsBundleDir').asFile.get()
+            resourcesDir = reactBundleTask.property('resourcesDir').asFile.get()
             jsBundleFile = file("$jsBundleDir/$bundleAssetName")
 
             generateBundledResourcesHash = tasks.create(
                     name: "generateBundledResourcesHash${targetName}",
                     type: Exec) {
                 commandLine (*nodeExecutableAndArgs, "${nodeModulesPath}/react-native-code-push/scripts/generateBundledResourcesHash.js", resourcesDir, jsBundleFile, jsBundleDir)
 
-                enabled config."bundleIn${targetName}" ||
-                config."bundleIn${variant.buildType.name.capitalize()}" ?:
-                targetName.toLowerCase().contains("release")
+                enabled !debuggableVariants.contains(variant.name) ?: targetName.toLowerCase().contains("release")
             }
-            
+
             runBefore("merge${targetName}Resources", generateBundledResourcesHash)
-            runBefore("merge${targetName}Assets", generateBundledResourcesHash)   
+            runBefore("merge${targetName}Assets", generateBundledResourcesHash)
```

Comment:
It looks like [there](https://github.com/microsoft/react-native-code-push/pull/2419/checks#step:16:432) is a build error due to duplicate assets

Reply:
@johnf Just a thought, would it be possible to run GitHub actions on your personal repo to verify what's wrong?

Reply:
@Protino Thanks for reminding me of the obvious!! :smile: trying that now

Reply:
@sesm were you switching between old and new versions? The old react native grade code and the new plugin seem to use different directories for flavours. debugUat vs debug/uat
So a clean is likely required in that instance

Reply:
@johnf I've been switching between old and new versions, but I've done `./gradlew clean` since then, so all artefacts should be cleaned. I'm having an issue right now, that after I create a build with release configuration (`./gradlew assemblePlaygroundRelease` in my case`), I have to clean the project before I can create the same build one more time, otherwise I get a `duplicate assets` error.

Reply:
I can resolve the issue `duplicate assets` by add these line to function `addJsBundleAndMetaToManifest` in `react-native-code-push/scripts/generateBundledResourcesHash.js` (under var savedResourcesManifestPath). The issue happen when start building project second time and we should remove `android/app/build/ASSETS/createBundleDebugJsAndAssets/CodePushHash` to make it works again

`             var savedResourcesManifestPath = assetsDir + "/" + CODE_PUSH_HASH_FILE_NAME;
+            if(fileExists(savedResourcesManifestPath)) {
+                fs.unlinkSync(savedResourcesManifestPath);
+            }`